### ISSUE TITLE
X11 server connection timed out because it was unreachable in v3.4.0

### ISSUE THREAD

**Issue by asasine** [1Password/arboard#153]
I see in #147 that the X11 connection timeout behavior was reverted, and this was released in v3.4.0, however I still get the error `Unknown error while interacting with the clipboard: X11 server connection timed out because it was unreachable` with this latest version when calling `Clipboard::new()`

**Comment by flyingpot** [comment]
The same error message on MacOS.

**Comment by complexspaces** [comment]
After more reading I see now that `spongebob` removed clipboard support. In that case, I greatly apologize for the trouble `arboard` caused you and your users.

After reviewing other downstream users who have linked to this issue, I believe the problem was _most likely_ caused by a combination of not enabling (even [at the time it was removed](https://github.com/asasine/spongebob/pull/23/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542L14) from your app) Arboard's `wayland-data-control` feature flag (which is needed for any Wayland support at all) and some compositors like GNOME's Mutter not supporting the Wayland protocol extensions `arboard` needs to function.

 If you ever decide to add optional clipboard support back to your app, a positive or negative signal on my hypothesis above would be appreciated.

Given that your app is no longer using `arboard` and the evidence from other downstream users that this is Wayland/X11 socket related in other cases, I'm going to close this as non-fixable since the current architecture of `arboard` is just incompatible with those Mutter-based Wayland servers and needs X11 as a fallback.

**Comment by Julien-cpsn** [comment]
Hello,

We are encountering the same issue with ATAC as seen here https://github.com/Julien-cpsn/ATAC/issues/173
Using Arch [PKGBUILD](https://gitlab.archlinux.org/archlinux/packaging/packages/atac/-/blob/main/PKGBUILD?ref_type=heads)

Do you have any idea on how to solve such a thing?

Have a great day!!

**Comment by complexspaces** [comment]
Hey there, sorry to hear you're still having problems.

Do you have any more context available on where the issue is appearing? Such as if the compositor is primarily Wayland or X11 and what the distribution (if any) is? 

These kind of issues are generally "desktop specific" so its hard to reproduce or further analyze without more information. But with that info, I'm happy to dig in more.

**Comment by orhun** [comment]
Hello. As referred above, the package is built from these [PKGBUILD instructions](https://gitlab.archlinux.org/archlinux/packaging/packages/atac/-/blob/main/PKGBUILD?ref_type=heads) and in a [clean chroot](https://wiki.archlinux.org/title/DeveloperWiki:Building_in_a_clean_chroot), so there isn't any compositor installed.

**Comment by complexspaces** [comment]
Oh sorry, I must have misunderstood the context reading through the issue you linked. You're saying that you are calling calling `arboard`'s functions while inside a `chroot` environment at package build time? I assume due to that there's no `/run` folder to find the X11 socket file in, or environment variables to point to a socket with?

**Comment by orhun** [comment]
Yeah, probably. That's why I initially suggested that arboard feature should be optional in https://github.com/Julien-cpsn/ATAC/issues/173
